src/domain/session/room/timeline/linkify.js

```diff
@@ -0,0 +1,12 @@
+export function linkify(text, callback) {
+    const regex = /(?:https|http|ftp):\/\/[a-zA-Z0-9:.\[\]#-]+(?:\/[^\s]*[^\s.,?!]|[^\s\u{80}-\u{10ffff}.,?!])/gui
```

Comment:
Can we have some comments explaining the different parts of the regex?
I'm also somewhat confused by:
 - the `#` in `[a-zA-Z0-9:.\[\]#-]`, hosts can't have a `#` in them AFAIK?
 - `(?:\/[^\s]*[^\s.,?!]|[^\s\u{80}-\u{10ffff}.,?!])`, `\/[^\s]` selects /* until space, `[^\s.,?!]` selects anything but whitespace and punctuation, and why do we need `[^\s\u{80}-\u{10ffff}.,?!]` again (something related to unicode but forgot why "not whitespace" doesn't cut it)? Also, what is the operator precedence of `|` here? E.g. what does it combine? `[^\s]*[^\s.,?!]` or `[^\s.,?!]`?